packages/transformers/js/src/JSTransformer.js

```diff
@@ -706,51 +706,67 @@ export default (new Transformer({
         asset.symbols.set(exported, local, convertLoc(loc));
       }
 
-      let deps = new Map(
-        asset
-          .getDependencies()
-          .map(dep => [dep.meta.placeholder ?? dep.specifier, dep]),
-      );
-      for (let dep of deps.values()) {
-        dep.symbols.ensure();
+      let deps = new Map();
+      for (let dep of asset.getDependencies()) {
+        if (!deps.has(dep.meta.placeholder ?? dep.specifier)) {
+          deps.set(dep.meta.placeholder ?? dep.specifier, []);
+        }
+        deps.get(dep.meta.placeholder ?? dep.specifier)?.push(dep);
```

Comment:
A possible alternative way of implementing this could be to make the map keyed by `(dep.meta.placeholder ?? dep.specifier) + dep.specifierType`, rather than an array of dependencies per specifier.

Reply:
I think the main reason why I didn't add the kind to the key is because we only want to do the `ImportKind` specific stuff when we are looking at `hoist_result.imported_symbols`. Keying by specifier + specifierType would mean that we would have to change all the other things in the `hoist_result` to track the `kind`, but I'm not sure if we should do that.